pkg/controller/v1alpha1/localmodelnode/controller.go

```
@@ -72,8 +74,43 @@ var (
 	fsHelper                   FileSystemInterface
 )
 
+// Returns the nodegroup of a node
+// NOTE: Assuming a node could only belong to 1 nodegroup
+func (c *LocalModelNodeReconciler) getNodeGroupFromNode(ctx context.Context, nodeName string) (*v1alpha1api.LocalModelNodeGroup, error) {
+	node := &v1.Node{}
+	if err := c.Get(ctx, types.NamespacedName{Name: nodeName}, node); err != nil {
+		return nil, err
+	}
+	nodeGroups := &v1alpha1api.LocalModelNodeGroupList{}
+	if err := c.List(ctx, nodeGroups); err != nil {
+		return nil, err
+	}
+	for _, nodeGroup := range nodeGroups.Items {
+		// To avoid implicit memory aliasing in for loop
+		nodeGroup := nodeGroup
+		matches, err := utils.CheckNodeAffinity(&nodeGroup.Spec.PersistentVolumeSpec, *node)
+		if err != nil {
+			return nil, err
+		}
+		if matches {
+			return &nodeGroup, nil
+		}
+	}
+	c.Log.Error(nil, "Did not find matching nodegroup for node", "node name", nodeName)
+
+	return nil, nil
```

should return an error